### PR TITLE
pkg: rest-hooks patch

### PR DESCRIPTION
**Pull request by ntucker**
## What
Upgrade rest-hooks to latest version (patch update)

## How
useStatefulResource() types were improved to be more precise. This correctly identified the possible undefined return value when using ternary of null in params. This will now type enforce, the already present optional chaining in these cases.

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

**Comment by michel-tricot** [comment]
Thank you @ntucker !

**Comment by cgardens** [comment]
Thanks @ntucker . Going to let @Jamakase review this since he's our resthooks expert.

File: airbyte-webapp/src/components/hooks/services/useDestinationHook.tsx [hunk 1]
```diff
@@ -26,7 +26,9 @@ export const useDestinationDefinitionSpecificationLoad = (
   destinationDefinitionId: string
 ): {
   isLoading: boolean;
-  destinationDefinitionSpecification: DestinationDefinitionSpecification;
+  destinationDefinitionSpecification:
```
Comment (Jamakase) [inline airbyte-webapp/src/components/hooks/services/useDestinationHook.tsx:29]: I suppose it's better to declare it as `destinationDefinitionSpecification?` as this is the pattern we are trying to follow everywhere.

**Review by jamakase — APPROVED** [review]

**Comment by ntucker** [comment]
@Jamakase updated this again with latest master and versions

**Comment by Jamakase** [comment]
> @Jamakase updated this again with latest master and versions

Hey @ntucker thank you for an update. Fast question - haven't you faced any issues after you updated fields to be nullable? I am quite sure that types will not match in some other components.

Maybe we should open separate issue to address these types if we can't change them right now? I understand that nullable type suits here better because during loading or on error we won't have an entity.

**Comment by ntucker** [comment]
@Jamakase I didn't get any type errors when on here besides those. I looked at some of the other cases and it looked like many of them already had the correct types like the cases I modified.

I believe `npm run build` runs typechecking. I also did `npm run test`

**Comment by ntucker** [comment]
@Jamakase i see you landed this and it broke master. If I can be of assistance, https://discord.gg/35nb8Mz we can chat there maybe? I can do slack too but it's harder to link to

**Comment by Jamakase** [comment]
> @Jamakase i see you landed this and it broke master. If I can be of assistance, https://discord.gg/35nb8Mz we can chat there maybe? I can do slack too but it's harder to link to

No worries @ntucker, I am already working on the proper fix :) But thank you for such a quick offer for help

**Pull request merged**